### PR TITLE
Migrate lighthouse.ts to TS

### PR DESCRIPTION
**Pull request by calebeby**

### CODE REVIEW

File: lighthouse.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ const runReport = async (url, outputFormat, maxWait) => {
     `--emulated-form-factor=mobile`,
     `--only-categories=performance`,
     `--chrome-flags="--headless"`,
-    `--max-wait-for-load=${maxWait || 45000}`,
```
Comment (emersonthis) [inline lighthouse.ts:17]: The thinking here was that it might be helpful for testing purposes to be able to override the timeout, presumably to cause it immediately and speed up the test. I can't remember if it's being used anywhere, yet...

To be clear, it was a workaround to compensate for bad architecture. The right way to do this is to actually mock the lighthouse logic in such a way that live never actually happen during testing. But I haven't gotten to figuring that out.

The bottom line is that I don't care if we remove this but if we do, we should have another strategy to replace what it was supposed to do.

File: lighthouse.ts [hunk 2]
```diff
@@ -1,11 +1,11 @@
-const spawnSync = require('child_process').spawnSync;
+import { spawnSync } from 'child_process';
+import sanitize from 'sanitize-filename';
+
 const lighthouseCli = require.resolve('lighthouse/lighthouse-cli');
```
Comment (emersonthis) [inline lighthouse.ts:4]: Out of curiosity, why is this one still `require`ed?

Comment (calebeby) [reply lighthouse.ts:4]: There isn't currently an equivalent for `require.resolve` in node with modules. Eventually it will support [`import.meta.resolve`](https://nodejs.org/api/esm.html#esm_no_require_resolve) but it doesn't yet.

We are currently transpiling from ES Modules => CommonJS, but TS (which does that transformation) doesn't support transpiling `import.meta.resolve` to `require.resolve`.

File: lighthouse.ts (hunk 1 above)
Comment (calebeby) [reply lighthouse.ts:17]: I took it out because the one place we are calling this function, we aren't passing in the parameter. I think it is easy enough to add it back if we end up using it in the future.

File: lighthouse.ts (hunk 2 above)
Comment (emersonthis) [reply lighthouse.ts:4]: Interesting

File: lighthouse.ts (hunk 1 above)
Comment (emersonthis) [reply lighthouse.ts:17]: Yea I'm remembering that I think the slow/unstable live network request problem is kind of temporarily "solved" by lack of test coverage for that code. 🤪

**Comment by emersonthis** [comment]
I can't figure out how to approve this after leaving comments so I'm just merging this myself...
